concrete/src/Http/Middleware/OAuthAuthenticationMiddleware.php

```diff
@@ -62,12 +67,20 @@ public function process(Request $request, DelegateInterface $frame)
         $request->attributes->add($newRequest->attributes->all());
 
         // Handle route scope
-        if ($route && $routeScopes = $route->getOption('oauth_scopes')) {
-            $requestScopes = $request->attributes->get('oauth_scopes');
+        if ($route) {
+            if ($routeScopes = $route->getOption('oauth_scopes')) {
+                $requestScopes = $request->attributes->get('oauth_scopes');
 
-            if (!array_intersect((array) $routeScopes, (array) $requestScopes)) {
-                throw new OAuthServerException(
-                    'Endpoint out of scope.', 1, 'invalid_request', 400, 'Try reauthorizing with the proper scope.'
+                if (!array_intersect((array)$routeScopes, (array)$requestScopes)) {
+                    throw new OAuthServerException(
+                        'Endpoint out of scope.', 1, 'invalid_request', 400, 'Try reauthorizing with the proper scope.'
+                    );
+                }
+            }
+
+            if ($userId = $request->attributes->get('oauth_user_id')) {
```

Comment:
I can't find the issue where this was but @mlocati argues that assignment in conditionals should be avoided since it's easy to get wrong

Reply:
Hey I'm just following the style of the request of the code in this class.